soc/riscv/riscv-privileged/hpmicro/soc.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2022 HPMicro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/__assert.h>
+#include <hpm_common.h>
+#include <hpm_soc.h>
+#include <hpm_pmp_drv.h>
+#include <hpm_pllctl_drv.h>
+#ifdef CONFIG_XIP
+#include <hpm_bootheader.h>
+#endif
+
+#ifdef CONFIG_XIP
+/**
+ * @brief FLASH configuration option definitions:
+ * option[0]:
```

Comment:
According to the description in the user manual:
this word should be Header, not option[0]